reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs

```diff
@@ -822,7 +913,7 @@ hoistTraverseWithKeyWithAdjust base mapPatch updateChildUnreadiness applyDomUpda
                   when (DMap.null newUnready) $ do
                     applyDomUpdate p
   (children0, children') <- ImmediateDomBuilderT $ lift $ base (\k v -> drawChildUpdate initialEnv markChildReady $ f k v) dm0 dm'
-  let processChild k (Compose (_, _, sRef, _)) = ComposeMaybe <$> do
+  let processChild k (_child_readyState . getCompose -> sRef) = ComposeMaybe <$> do
```

Comment:
Is this the only use of `ViewPatterns` in the file?  Although I don't have a strong preference against `ViewPatterns`, I think in this particular case it might be just as good to do something like `Compose (Child { _child_readyState = sRef })`